### PR TITLE
Update to tagged ITK 5.1.1 version

### PR DESCRIPTION
**Pull request by jamesobutler**
This updates the ITK version that Slicer uses to an official ITK tagged version (5.1.1).  I first ran into a build error related to TBB that was seemingly the result of the change in https://github.com/InsightSoftwareConsortium/ITK/commit/51792db03b24d1aa7e80524998064c1745fe4d99. The version of TBB currently specified in the BRAINSTools (the Slicer version) superbuild is actually a version of v2020.2 as seen [here](https://github.com/Slicer/BRAINSTools/blob/slicer-2020-04-14-v5.2.0/SuperBuild/External_TBB.cmake#L27) so I expect we can update the version to at least that high in the future without issues with BRAINSTools. I also changed to download the TBB binaries from the [official TBB github repo](https://github.com/oneapi-src/oneTBB) instead of the [Slicer/SlicerBinaryDependencies](https://github.com/Slicer/SlicerBinaryDependencies) repo. I'm guessing the TBB binaries were not previously offered in a easy public way on github before or that the binaries were not built with a compatible compiler for the Slicer factory build machine?

I updated the TBB version to 2019_U9 because the minimum mac OS X version for this update is 10.11 ([source](https://github.com/oneapi-src/oneTBB/blob/2019_U9/doc/Release_Notes.txt)). Starting with TBB v2020.0 the minimum macOS version is 10.14 ([source](https://github.com/oneapi-src/oneTBB/blob/v2020.0/doc/Release_Notes.txt)) which is greater than the current Slicer minimum macOS version of 10.13. Also starting with TBB version v2020.0 it drops Windows 8.1 and Windows 7 support and requires the Visual Studio 2017 compiler or later.

I've updated SimpleITK to v2.0rc3 which is using a more recent ITK 5.1.x version than we were previously using.  I've specified the SimpleITK upstream repo because there are currently no Slicer specific patches needed.  If using the [Slicer/SimpleITK](https://github.com/Slicer/SimpleITK) fork is still desired, I can change it back and push a new branch to the fork based on the new git commit hash. I'm expecting an update to using a SimpleITK official 2.0.0 tag in the near future.

Currently building in release mode on Windows 10 with standard configuration except for enabling parallel build. Will report results here once finished.

### CODE REVIEW

**Comment by jamesobutler** [comment]
Currently running into the following build errors
```
45>    SimpleITKPYTHON_wrap.cxx
45>C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITKPYTHON_wrap.cxx(55545,3): error C2065: 'PixelIDValueType': undeclared identifier [C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITK_PYTHON.vcxproj] [C:\S5R-nightly\SimpleITK-build\SimpleITK.vcxproj]
45>C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITKPYTHON_wrap.cxx(55545,20): error C2146: syntax error: missing ';' before identifier 'result' [C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITK_PYTHON.vcxproj] [C:\S5R-nightly\SimpleITK-build\SimpleITK.vcxproj]
45>C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITKPYTHON_wrap.cxx(55545,20): error C2065: 'result': undeclared identifier [C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITK_PYTHON.vcxproj] [C:\S5R-nightly\SimpleITK-build\SimpleITK.vcxproj]
45>C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITKPYTHON_wrap.cxx(55563,9): error C2065: 'result': undeclared identifier [C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITK_PYTHON.vcxproj] [C:\S5R-nightly\SimpleITK-build\SimpleITK.vcxproj]
45>C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITKPYTHON_wrap.cxx(55582,15): error C2061: syntax error: identifier 'PixelIDValueType' [C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITK_PYTHON.vcxproj] [C:\S5R-nightly\SimpleITK-build\SimpleITK.vcxproj]
45>C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITKPYTHON_wrap.cxx(55582,15): error C2660: 'SWIG_Python_NewPointerObj': function does not take 1 arguments [C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITK_PYTHON.vcxproj] [C:\S5R-nightly\SimpleITK-build\SimpleITK.vcxproj]
45>  C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITKPYTHON_wrap.cxx(2628,1): message : see declaration of 'SWIG_Python_NewPointerObj' [C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITK_PYTHON.vcxproj]
45>C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITKPYTHON_wrap.cxx(56207,3): error C2065: 'PixelIDValueType': undeclared identifier [C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITK_PYTHON.vcxproj] [C:\S5R-nightly\SimpleITK-build\SimpleITK.vcxproj]
45>C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITKPYTHON_wrap.cxx(56207,20): error C2146: syntax error: missing ';' before identifier 'result' [C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITK_PYTHON.vcxproj] [C:\S5R-nightly\SimpleITK-build\SimpleITK.vcxproj]
45>C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITKPYTHON_wrap.cxx(56207,20): error C2065: 'result': undeclared identifier [C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITK_PYTHON.vcxproj] [C:\S5R-nightly\SimpleITK-build\SimpleITK.vcxproj]
45>C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITKPYTHON_wrap.cxx(56219,9): error C2065: 'result': undeclared identifier [C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITK_PYTHON.vcxproj] [C:\S5R-nightly\SimpleITK-build\SimpleITK.vcxproj]
45>C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITKPYTHON_wrap.cxx(56238,15): error C2061: syntax error: identifier 'PixelIDValueType' [C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITK_PYTHON.vcxproj] [C:\S5R-nightly\SimpleITK-build\SimpleITK.vcxproj]
45>C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITKPYTHON_wrap.cxx(56238,15): error C2660: 'SWIG_Python_NewPointerObj': function does not take 1 arguments [C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITK_PYTHON.vcxproj] [C:\S5R-nightly\SimpleITK-build\SimpleITK.vcxproj]
45>  C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITKPYTHON_wrap.cxx(2628,1): message : see declaration of 'SWIG_Python_NewPointerObj' [C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITK_PYTHON.vcxproj]
45>C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITKPYTHON_wrap.cxx(83319,3): error C2065: 'PixelIDValueType': undeclared identifier [C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITK_PYTHON.vcxproj] [C:\S5R-nightly\SimpleITK-build\SimpleITK.vcxproj]
45>C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITKPYTHON_wrap.cxx(83319,20): error C2146: syntax error: missing ';' before identifier 'result' [C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITK_PYTHON.vcxproj] [C:\S5R-nightly\SimpleITK-build\SimpleITK.vcxproj]
45>C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITKPYTHON_wrap.cxx(83319,20): error C2065: 'result': undeclared identifier [C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITK_PYTHON.vcxproj] [C:\S5R-nightly\SimpleITK-build\SimpleITK.vcxproj]
45>C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITKPYTHON_wrap.cxx(83331,9): error C2065: 'result': undeclared identifier [C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITK_PYTHON.vcxproj] [C:\S5R-nightly\SimpleITK-build\SimpleITK.vcxproj]
45>C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITKPYTHON_wrap.cxx(83350,15): error C2061: syntax error: identifier 'PixelIDValueType' [C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITK_PYTHON.vcxproj] [C:\S5R-nightly\SimpleITK-build\SimpleITK.vcxproj]
45>C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITKPYTHON_wrap.cxx(83350,15): error C2660: 'SWIG_Python_NewPointerObj': function does not take 1 arguments [C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITK_PYTHON.vcxproj] [C:\S5R-nightly\SimpleITK-build\SimpleITK.vcxproj]
45>  C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITKPYTHON_wrap.cxx(2628,1): message : see declaration of 'SWIG_Python_NewPointerObj' [C:\S5R-nightly\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITK_PYTHON.vcxproj]
45>Done building project "SimpleITK.vcxproj" -- FAILED.
```
```
47>  vtkITKMorphologicalContourInterpolator.cxx
47>     Creating library C:/S5R-nightly/Slicer-build/lib/Slicer-4.11/Release/vtkITK.lib and object C:/S5R-nightly/Slicer-build/lib/Slicer-4.11/Release/vtkITK.exp
47>vtkITKArchetypeImageSeriesVectorReaderSeries.obj : error LNK2019: unresolved external symbol "public: __cdecl gdcm::Global::Global(void)" (??0Global@gdcm@@QEAA@XZ) referenced in function "void __cdecl gdcm::`dynamic initializer for 'GlobalInstance''(void)" (??__EGlobalInstance@gdcm@@YAXXZ) [C:\S5R-nightly\Slicer-build\Libs\vtkITK\vtkITK.vcxproj]
47>vtkITKArchetypeImageSeriesVectorReaderSeries.obj : error LNK2019: unresolved external symbol "public: __cdecl gdcm::Global::~Global(void)" (??1Global@gdcm@@QEAA@XZ) referenced in function "void __cdecl gdcm::`dynamic atexit destructor for 'GlobalInstance''(void)" (??__FGlobalInstance@gdcm@@YAXXZ) [C:\S5R-nightly\Slicer-build\Libs\vtkITK\vtkITK.vcxproj]
47>C:\S5R-nightly\Slicer-build\bin\Release\vtkITK.dll : fatal error LNK1120: 2 unresolved externals [C:\S5R-nightly\Slicer-build\Libs\vtkITK\vtkITK.vcxproj]
```

**Comment by jamesobutler** [comment]
I've issued an additional commit to remove the gdcmGlobal.h includes within vtkITKArchetypeImageSeriesVectorReaderSeries.cxx which solves the previous Slicer build errors mentioned.  I ran tests on Windows and the same tests that are currently failing were also the ones failing, plus SimpleITK related tests since I haven't resolved its build errors.  Therefore I think it should be safe to remove this includes to fix the build error.

@blowekamp Do you know what is up with the SimpleITK python wrapping errors listed in the previous post?

File: SuperBuild/External_SimpleITK.cmake [hunk 1]
```diff
@@ -45,13 +45,13 @@ ExternalProject_Execute(${proj} \"install\" \"${PYTHON_EXECUTABLE}\" Packaging/s
```
Comment (blowekamp) [inline SuperBuild/External_SimpleITK.cmake:45]: Change the above to: 
```
ExternalProject_Execute(${proj} \"install\" \"${PYTHON_EXECUTABLE}\" setup.py install)
```

**Comment by blowekamp** [comment]
Great to see Slicer being on top of the SimpleITK 2.0rc3 tagged on Friday. Please pay attention to the ITK discourse for the posting of the release notes later this week.

SWIG compilation error are likely related to the old 3.0 version of SWIG. Please update to [SWIG 4.0.1](https://github.com/SimpleITK/SimpleITK/blob/master/SuperBuild/External_Swig.cmake), and [PCRE](https://github.com/SimpleITK/SimpleITK/blob/master/SuperBuild/External_PCRE.cmake) could use an update too ( but that would be causing the problem ).

ITK will need to be configure with the following variable defined:
```
Module_SimpleITKFilters_GIT_TAG=ce51d77771a54e7e0791fadb15e50dc38cbd8358
```
This updates the git hash used for the SimpleITKFilters remote module.

In SimpleITK 2.0rc2 incompatible changes are made. They were listed in the release notes:
https://github.com/SimpleITK/SimpleITK/releases/tag/v2.0rc2

This is anticipated to be the last RC before SimpleITK 2.0 is released.

**Comment by jcfr** [comment]
Thanks @blowekamp  for the detailed notes :pray:

**Comment by jamesobutler** [comment]
@jcfr I don't have the permissions to create new releases at https://github.com/Slicer/SlicerBinaryDependencies/releases. You can either do this or I can with increased permissions.

What should be created would be like:
- "PCRE_8.44" with asset: https://ftp.pcre.org/pub/pcre/pcre-8.44.tar.gz (latest version of the original PCRE implementation)
- "SWIG_4.0.2" with assets: https://github.com/swig/swig/archive/v4.0.2.tar.gz and http://prdownloads.sourceforge.net/swig/swigwin-4.0.2.zip (latest version and the version specified in the SimpleITK repo)

**Comment by jcfr** [comment]
> You can either do this or I can with increased permissions.

@jamesobutler I just added you to [slicer-core](https://github.com/orgs/Slicer/teams/slicer-core/members) team, could you confirm you have access ?

**Comment by jamesobutler** [comment]
Instead of creating new releases in https://github.com/Slicer/SlicerBinaryDependencies/releases for updated versions of SWIG and PCRE, I have followed the convention in the [TCL release](https://github.com/Slicer/SlicerBinaryDependencies/releases/tag/tcl) to include new versions as additional assets to that release.

I will continue with the updating of the Slicer build process to use these updated release assets.

**Comment by jamesobutler** [comment]
@jcfr Would you like the update intel tbb added to https://github.com/Slicer/SlicerBinaryDependencies/releases/tag/tbb to use it from there or do you think it is okay to get these assets from the official github intel repo release as I did in my TBB update commit for this PR?

**Comment by jcfr** [comment]
> Would you like the updated intel tbb added to https://github.com/Slicer/SlicerBinaryDependencies/releases/tag/tbb to use it from there or do you think it is okay to get these assets from the official github intel tbb repo release as I did in my TBB update commit for this PR?

Since the TBB release are now officially served from GitHub, I have no problem deprecating the release https://github.com/Slicer/SlicerBinaryDependencies/releases/tag/tbb

Could you update the release description indicating it is deprecated and referencing this discussion ?  :pray:

**Comment by jamesobutler** [comment]
This PR is officially ready for review now. Using Windows 10, there are no build errors with the standard Slicer build configuration and the tests that failed were the same as the ones that are already failing. 

cc: @pieper as a potential review to test the build on macOS for a sanity check

**Comment by jcfr** [comment]
Thanks  @jamesobutler 

I just rebased  against master.

* Linux build in progress

**Comment by jcfr** [comment]
Here is an errror reprorted on Linux:

```
[ 90%] Performing configure step for 'Swig'
-- Swig: Removing 'configure' log files
-- Swig: Swig_WORKING_DIR: /home/jcfr/Projects/Slicer-Release/Swig-build
-- Swig: sh;/home/jcfr/Projects/Slicer-Release/Swig/configure;--prefix=/home/jcfr/Projects/Slicer-Release/Swig-install;--with-pcre-prefix=/home/jcfr/Projects/Slicer-Release/PCRE-install;--without-octave;--with-python=/home/jcfr/Projects/Slicer-Release/python-install/bin/PythonSlicer
-- Swig: Errors detected - See below.

sh: 0: Can't open /home/jcfr/Projects/Slicer-Release/Swig/configure

CMake Error at /home/jcfr/Projects/Slicer/CMake/ExternalProjectForNonCMakeProject.cmake:104 (message):
  Swig: configure step failed with exit code '127'.

  Outputs also captured in
  /home/jcfr/Projects/Slicer-Release/Swig_configure_step_output.txt and
  /home/jcfr/Projects/Slicer-Release/Swig_configure_step_error.txt.

  Setting env.  variable EP_EXECUTE_DISABLE_CAPTURE_OUTPUTS to 1 allows to
  disable file capture.

Call Stack (most recent call first):
  /home/jcfr/Projects/Slicer-Release/Swig_configure_step.cmake:3 (ExternalProject_Execute)
```

**Comment by jcfr** [comment]
It turns out the official source distribution (http://prdownloads.sourceforge.net/swig/swig-4.0.2.tar.gz) should be used, I am updating `SlicerBinaryDependencies`

**Comment by jcfr** [comment]
* macOS build in progress

**Comment by jcfr** [comment]
* linux build :heavy_check_mark: , package creation :heavy_check_mark:

**Comment by jcfr** [comment]
* macOS build (on factory-south-macos) :heavy_check_mark:  , package creation :hourglass_flowing_sand:

**Comment by jamesobutler** [comment]
@jcfr So all platforms appear to have been successful?

**Comment by jcfr** [comment]
Look like there are macOS packaging issues associated to SimpleITK:

```
[...]
5 - fixing item /Volumes/D/T/S-b/Slicer-build/_CPack_Packages/macosx-amd64/DragNDrop/Slicer-4.11.0-2020-08-25-macosx-amd64/Slicer.app/Contents/lib/Python/lib/python3.6/site-packages/SimpleITK-2.0.0rc3-py3.6-macosx-10.13-x86_64.egg/SimpleITK/_SimpleITK.cpython-36m-darwin.so with @rpath/lib/Python/lib/python3.6/site-packages/SimpleITK-2.0.0rc3-py3.6-macosx-10.13-x86_64.egg/SimpleITK/
warning: target '@rpath/libSimpleITKBasicFilters1-2.0.1.dylib' is not absolute...
warning: target '@rpath/libSimpleITKBasicFilters1-2.0.1.dylib' does not exist...
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/objdump: '@rpath/libSimpleITKBasicFilters1-2.0.1.dylib': No such file or directory
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/objdump: '@rpath/libSimpleITKBasicFilters1-2.0.1.dylib': No such file or directory
[...]
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/install_name_tool: can't open file: /Volumes/D/T/S-b/Slicer-build/_CPack_Packages/macosx-amd64/DragNDrop/Slicer-4.11.0-2020-08-25-macosx-amd64/Slicer.app/Contents/lib/Slicer-4.11/libSimpleITK_ITKWatersheds-2.0.1.dylib (No such file or directory)
warning: target '/Volumes/D/T/S-b/Slicer-build/_CPack_Packages/macosx-amd64/DragNDrop/Slicer-4.11.0-2020-08-25-macosx-amd64/Slicer.app/Contents/lib/Slicer-4.11/libSimpleITK_SimpleITKFilters-2.0.1.dylib' does not exist...
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/objdump: '/Volumes/D/T/S-b/Slicer-build/_CPack_Packages/macosx-amd64/DragNDrop/Slicer-4.11.0-2020-08-25-macosx-amd64/Slicer.app/Contents/lib/Slicer-4.11/libSimpleITK_SimpleITKFilters-2.0.1.dylib': No such file or directory
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/objdump: '/Volumes/D/T/S-b/Slicer-build/_CPack_Packages/macosx-amd64/DragNDrop/Slicer-4.11.0-2020-08-25-macosx-amd64/Slicer.app/Contents/lib/Slicer-4.11/libSimpleITK_SimpleITKFilters-2.0.1.dylib': No such file or directory
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/install_name_tool: can't open file: /Volumes/D/T/S-b/Slicer-build/_CPack_Packages/macosx-amd64/DragNDrop/Slicer-4.11.0-2020-08-25-macosx-amd64/Slicer.app/Contents/lib/Slicer-4.11/libSimpleITK_SimpleITKFilters-2.0.1.dylib (No such file or directory)
[...]
818 - fixing item /Volumes/D/Support/qt-everywhere-build-5.15.0/lib/QtXml.framework/Versions/5/QtXml with @rpath/Frameworks
CPack: Create package
CPack: - package: /Volumes/D/T/S-b/Slicer-build/Slicer-4.11.0-2020-08-25-macosx-amd64.dmg generated.

```

**Comment by jamesobutler** [comment]
@jcfr And there was no build errors with SimpleITK on macOS?
@blowekamp Do these packaging errors on macOS look familiar to you in anyway?

Is this some issue using rpath? Or is it truly missing some files?

**Comment by jcfr** [comment]
Given the current macOS packaging process, we need to make sure the SimpleITK libraries have a complete path (meaning no `@rpath`)

Prior to the SimpleITK updates done in this PR, there is only one SimpleITK python module that depends on ITK liraries:

```
$ otool -L ./lib/Python/lib/python3.6/site-packages/SimpleITK-2.0.0rc2.dev133-py3.6-macosx-10.13-x86_64.egg/SimpleITK/_SimpleITK.cpython-36m-darwin.so
./lib/Python/lib/python3.6/site-packages/SimpleITK-2.0.0rc2.dev133-py3.6-macosx-10.13-x86_64.egg/SimpleITK/_SimpleITK.cpython-36m-darwin.so:
	@rpath/lib/Slicer-4.11/libITKIOBruker-5.1.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/lib/Slicer-4.11/libITKIODCMTK-5.1.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/lib/Slicer-4.11/libITKIOHDF5-5.1.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/lib/Slicer-4.11/libITKIOJPEG2000-5.1.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/lib/Slicer-4.11/libITKIOLSM-5.1.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	[...]
	@rpath/lib/Slicer-4.11/libitknetlib-5.1.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/lib/Slicer-4.11/libitkvcl-5.1.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.200.5)
	@rpath/lib/Slicer-4.11/libITKPath-5.1.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 400.9.4)

```

It doesn't depend on libraries like `libSimpleITKBasicFilters1-2.0.1.dylib`

It looks like disabling shared build on unix does not have the intended effect anymore: 

https://github.com/Slicer/Slicer/blob/fe9579f0689e57c89fc908b2f9d79f75066047ba/SuperBuild/External_SimpleITK.cmake#L96

https://github.com/Slicer/Slicer/blob/fe9579f0689e57c89fc908b2f9d79f75066047ba/CMakeLists.txt#L343-L358

**Comment by jcfr** [comment]
In the build associated with this PR, here are the dependency of SimpleITK in the build tree:

```
$ otool -L ./SimpleITK-2.0.0rc3-py3.6-macosx-10.13-x86_64.egg/SimpleITK/_SimpleITK.cpython-36m-darwin.so
./SimpleITK-2.0.0rc3-py3.6-macosx-10.13-x86_64.egg/SimpleITK/_SimpleITK.cpython-36m-darwin.so:
	@rpath/libSimpleITKBasicFilters1-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITKIO-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITKRegistration-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKAnisotropicSmoothing-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKAntiAlias-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKBiasCorrection-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKBinaryMathematicalMorphology-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKClassifiers-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKColormap-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKCommon-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKConnectedComponents-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKConvolution-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKCurvatureFlow-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKDeconvolution-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKDenoising-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKDisplacementField-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKDistanceMap-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKFFT-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKFastMarching-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKImageCompare-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKImageCompose-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKImageFeature-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKImageFilterBase-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKImageFunction-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKImageFusion-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKImageGradient-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKImageGrid-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKImageIntensity-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKImageLabel-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKImageNoise-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKImageSources-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKImageStatistics-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKLabelMap-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKLabelVoting-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKLevelSets-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKMathematicalMorphology-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKPDEDeformableRegistration-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKRegionGrowing-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKRegistrationCommon-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKReview-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKSmoothing-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKSuperPixel-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKThresholding-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKTransform-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_ITKWatersheds-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITK_SimpleITKFilters-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITKBasicFilters0-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/libSimpleITKCommon-2.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 400.9.4)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.200.5)

```

**Comment by jamesobutler** [comment]
Maybe because if you have shared libs off here
https://github.com/Slicer/Slicer/blob/fe9579f0689e57c89fc908b2f9d79f75066047ba/SuperBuild/External_SimpleITK.cmake#L96

that same setting isn't also set to ITK, where we are always specifying shared libs.
https://github.com/Slicer/Slicer/blob/fe9579f0689e57c89fc908b2f9d79f75066047ba/SuperBuild/External_ITK.cmake#L139

SimpleITK's superbuild of ITK matches shared lib usage of SimpleITK with the ITK build.
https://github.com/SimpleITK/SimpleITK/blob/v2.0rc3/SuperBuild/External_ITK.cmake#L69-L72

**Comment by blowekamp** [comment]
Was this a local build? Would the build happened to have been in Debug mode?

**Comment by jcfr** [comment]
> Was this a local build? Would the build happened to have been in Debug mode?

Good catch. That was indeed a debug. I am restarting a build in Release.

**Comment by jcfr** [comment]
Here is an issue for improving the build system and more quickly detect similar situation.
See https://github.com/Slicer/Slicer/issues/5145

**Comment by jcfr** [comment]
I confirm macOS packaging works as expected.

**Pull request merged**